### PR TITLE
Add a patch to insert a static cast to the argument of std::abs

### PR DESCRIPTION
**Pull request by datalogics-robb**
Add a patch to insert a static cast to the argument of std::abs to make IBM's xlclang++ happy.

Fixes issue #4312 

Specify library name and version:  **catch2/2.13.4**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/catch2/2.x.x/conanfile.py [hunk 1]
```diff
@@ -49,7 +49,13 @@ def _configure_cmake(self):
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            print("aplying patch %s" % (patch,))
```
Comment (uilianries) [inline recipes/catch2/2.x.x/conanfile.py:54]:
```suggestion
```

Not needed.

When you really need to print something, use `self.output.info` instead.

**Review by uilianries — APPROVED** [review]

**Comment by prince-chrismc** [comment]
As per #3903 I think we should wait to see if upstream accepts the patch before merging... this way it reduces our overhead.

https://github.com/catchorg/Catch2/pull/2155

**Comment by uilianries** [comment]
Usually I'm not opposite of building patches, but @prince-chrismc comments makes sense. Catch2 releases new versions frequently and I think we could wait for a next one if they have a close date.

**Comment by datalogics-robb** [comment]
@prince-chrismc I agree, getting the patch merged upstream is important, and reduces CCI overhead.  However, it will not fix the current version, which is why I also produced *this* PR.  It's such a trivial change and clearly follows the patching guidelines.  I could have patched older versions, but if a user needs this patch they can update to 2.13.4, so it didn't seem worthwhile.  Would you like that done?

**Comment by prince-chrismc** [comment]
This is what I wanted to see

> horenmar merged commit 68975e3 into catchorg:v2.x 36 minutes ago
1 of 2 checks passed

This is absolute a build patch so it's acceptable, but I am no expert to say it's trivial which is why I wanted to delay to upstreams reaction.

**Review by prince-chrismc — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Pull request merged**